src/base/server_type.cpp

```diff
@@ -57,15 +57,24 @@ void set_server_type (SERVER_TYPE type)
 int init_server_type (const char *db_name)
 {
   int er_code = NO_ERROR;
+  SERVER_TYPE parameter_value = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
   if (g_server_type == SERVER_TYPE_UNKNOWN)
     {
-      g_server_type = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
+      if (parameter_value == SERVER_TYPE_SINGLE_NODE)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INVALID_SERVER_OPTION, 1,
+		  "Single node server must have type specified as argument");
+	  return ER_INVALID_SERVER_OPTION;
+	}
+      g_server_type = parameter_value;
```

Comment:
Currently, SERVER_TYPE_TRANSACTION is the default value. So, I think the following if statement is not necessary.